### ISSUE TITLE
[Timelines] Bring over CSS for Timelines, Clocks and Timers

### ISSUE THREAD

**Issue by VWoeltjen** [nasa/openmctweb#208]
#207 brings timelines into Open MCT Web, but does not include CSS necessary to display timelines correctly. Bringing over CSS appears non-trivial, as it includes both styles for timelines (want), and styles to override platform look-and-feel (don't want)

Goal here is to get the CSS classes that timeline needs into the open source repo, either in `platform/features/timeline` or in `platform/commonUI/general` (as appropriate)

Should also be able to add `platform/features/timeline` to `bundles.json` to re-enable it at the conclusion of this issue.

**Comment by charlesh88** [comment]
@VWoeltjen  I really like the idea that significant features (like Timelines) will only contribute their CSS when their bundle is loaded. This avoids bloating out the main espresso and snow css theme files, and supports a more modular architecture. But, there's a few downsides:
1. Bundle-based SASS files will need to refer to main platform and theme-based constants files in order to use theme values properly, and to create their own theme files. This means that changes to the core SASS could get out of sync with a feature bundle, and the next time the feature bundle goes to compile it errors because something changed in core.
2. In order to support theming, bundles.json would have to load a separate bundle for the desired theme for a given feature bundle. For Timelines for example, bundles.json would have to load
`platform/features/timeline,`
`platform/features/timeline/espresso`
This isn't terrible, but it's a little messy: if the theme wasn't loaded explicitly, Timelines wouldn't have a default fallback and would look bad. I'm not seeing a way around this right now.

So, process:
1. Checkout -b open208 from github/open1239. Do safety merge of latest github/master.
2. Copy the CSS code from /warp/... and place such that the `/platform/features/timeline/&lt;theme name&gt;` bundle will include its own SASS directory and rendered .css file. Add that bundle to bundles.json.
3. Load `/plaform/features/timeline` and make sure it looks Ok.
4. Push work to open208 and file a pull request against master.
5. File a new issue in JIRA for WARP to modify /warp/.../sass to remove the items that were moved in #208. Checkout a warp topic branch from origin/warp-master and merge in open208. **IMPORTANT: would it be better to wait until the pull request on open208 is merged into master? This is a case where I don't think so, the changes are fairly cut and dried - it's just moving some files around.** Do the work in  /warp* specifically with no changes to open208. If changes are needed, make them there, then merge back into the warp topic branch. Push when done to origin.
6. File a JIRA review and integrate issue for the warp topic branch.

Take this approach for major features moving forward. Consider doing this for Clocks and Timers bundles.

**Comment by charlesh88** [comment]
Working on this now. Pushed starting files to open208 branch.

**Comment by VWoeltjen** [comment]
Reassigning to me to give feedback on comment above, need to approach for themes.

@larkin recommends going to something standard like grunt/gulp to have a single build process for CSS

**Comment by VWoeltjen** [comment]
@charlesh88 - does the proposed solution to #241 makes sense here? If I were to make those changes, timeline CSS would look like:

```json
"stylesheets": [
    {
        "stylesheetUrl": "css/timeline.css"
    },
    {
        "stylesheetUrl": "css/timeline-snow.css",
        "theme": "snow"
    },
    {
        "stylesheetUrl": "css/timeline-espresso.css",
        "theme": "espresso"
    }
]
```

Where `timeline.css` is the common CSS (if any), and the others are the theme-specific files.

Would this throw off SASS compilation or create other issues?

This would change process to (changes in bold):

1. Checkout -b open208 from github/open1239. Do safety merge of latest github/master.
2. Copy the ~~CSS~~ __SASS__ code from /warp/... and place such that the __/platform/features/timeline__ bundle will include its own SASS directory and rendered .css files. Add that bundle to bundles.json.
3. Load /plaform/features/timeline and make sure it looks Ok.
4. Push work to open208 and file a pull request against master.
5. File JIRA issue to remove bundles from warp-master now that they've been transitioned

**Comment by charlesh88** [comment]
Yes, and just added comments and questions in the issue.

**Comment by charlesh88** [comment]
Currently blocked by #242

**Comment by charlesh88** [comment]
#242 has been merged; will proceed on this in Banks.

**Comment by charlesh88** [comment]
Amended title to include Clocks and Timers - they don't support the Snow theme yet.

**Comment by charlesh88** [comment]
@VWoeltjen I've pushed my final styling work on this to open208. Before I file a pull request, would like someone to take at some problems I've seen with Timelines: 
1. Dragging swimlane items activities isn't working
1. Tabular isn't displaying time properly
1. The zoom buttons don't work, probably related to 2.
I've not filed separate issues for this. Here's some screenshots of what I see:
![screen shot 2015-11-20 at 4 24 16 pm](https://cloud.githubusercontent.com/assets/1056412/11315654/90571384-8faa-11e5-974f-8b804dcff316.png)
![screen shot 2015-11-20 at 4 23 47 pm](https://cloud.githubusercontent.com/assets/1056412/11315661/a2632f5e-8faa-11e5-856f-12398cd162aa.png)

If these problems can be resolved and no issues come up related to look-and-feel, work in this branch is good to go to be integrated. 
Reassigning to Victor to have a look at this.

**Comment by VWoeltjen** [comment]
Pushed branch open208b with a fix for this, you should be able to merge this into open208 and complete from there.

Issue was a naming collision; I'd removed the `warp` prefix on various extensions when bringing over to open source, which inadvertently caused timelines/activities to use the wrong date-time control in there forms. Added a `timeline-` prefix for this control to disambiguate.

**Comment by VWoeltjen** [comment]
> If these problems can be resolved and no issues come up related to look-and-feel, work in this branch is good to go to be integrated. 

Ah, missed this comment - will complete this merge now.

**Comment by RCarvalhoNASA** [comment]
Testathon 20151123:  This all seems to be working correctly now
